connectors/sources/google_drive.py

```diff
@@ -0,0 +1,638 @@
+                os.path.abspath(DEFAULT_PEM_FILE)
+            ).read(),  # TODO: change this and provide meaningful defaults
+            "client_email": "123-abc@developer.gserviceaccount.com",
+            "client_id": "123-abc.apps.googleusercontent.com",
+            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+            "token_uri": "http://localhost:10339/token",
+        }
+        return {
+            "service_account_credentials": {
+                "display": "textarea",
+                "label": "Google Drive service account JSON",
+                "order": 1,
+                "type": "str",
+                "value": json.dumps(default_credentials),
+            },
+            "retry_count": {
+                "default_value": DEFAULT_RETRY_COUNT,
+                "display": "numeric",
+                "label": "Maximum retries for failed requests",
+                "order": 2,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": DEFAULT_RETRY_COUNT,
+            },
+        }
+
+    @cached_property
+    def _google_drive_client(self):
+        """Initialize and return the GoogleDriveClient
+
+        Returns:
+            GoogleDriveClient: An instance of the GoogleDriveClient.
+        """
+        json_credentials = json.loads(self.configuration["service_account_credentials"])
+
+        if (
+            json_credentials.get("private_key")
+            and "\n" not in json_credentials["private_key"]
+        ):
+            json_credentials["private_key"] = get_pem_format(
+                key=json_credentials["private_key"].strip(),
+                max_split=2,
+            )
+
+        return GoogleDriveClient(
+            json_credentials=json_credentials,
+            retry_count=self.configuration["retry_count"],
+        )
+
+    async def ping(self):
+        """""Verify the connection with Google Drive""" ""
+        if RUNNING_FTEST:
+            return
+
+        try:
+            await anext(
+                self._google_drive_client.api_call(
+                    resource="about", method="get", fields="kind"
+                )
+            )
+            logger.info("Successfully connected to the Google Drive.")
+        except Exception:
+            logger.exception("Error while connecting to the Google Drive.")
+            raise
+
+    async def validate_config(self):
+        """Validates whether user inputs are valid or not for configuration field.
+
+        Raises:
+            Exception: The format of service account json is invalid.
+        """
+        self.configuration.check_valid()
+
+        try:
+            json.loads(self.configuration["service_account_credentials"])
+        except ValueError as e:
+            raise ConfigurableFieldValueError(
+                "Google Drive service account is not a valid JSON."
+            ) from e
+
+    async def get_drives(self):
```

Comment:
Do you think it makes sense to make actually rich client with these methods? E.g. your client will have methods `get_drives`, `retrieve_all_drives` and such and the connector class itself will contain only the logic to call these methods and create view models, etc